src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -104,31 +120,24 @@ protected Optional<TokenPermissions> getTokenPermissions(HttpServletRequest requ
         return AuthorisationUtil.getTokenPermissions(request);
     }
 
-    private boolean doRoeUpdateJourneyChecks(HttpServletRequest request, HttpServletResponse response, String reqId, String companyNumberInScope,
-                                             String transactionId, Map<String, Object> authInfoMap) {
-        boolean isValidOeNumber = companyNumberInScope.startsWith("OE");
-        if (isValidOeNumber) {
-            boolean doCompanyNumbersMatch = doCompanyNumbersMatch(request, response, companyNumberInScope, transactionId);
-            authInfoMap.put("roe_update_journey_company_numbers_match", doCompanyNumbersMatch);
-            if (doCompanyNumbersMatch) {
-                ApiLogger.debugContext(reqId, "UserAuthenticationInterceptor authorised with company_oe_annual_update=create permission",
-                        authInfoMap);
-                return true;
-            } else {
-                ApiLogger.errorContext(reqId, "UserAuthenticationInterceptor unauthorised for ROE Update Journey as Company Numbers inScope and Transaction do not match", null, authInfoMap);
-                return false;
-            }
-        } else {
-            ApiLogger.debugContext(reqId, "UserAuthenticationInterceptor unauthorised for ROE Update Journey as Company Number is not a valid OE",
-                    authInfoMap);
-            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
-            return false;
+    private boolean isRequestRoeRegistration(Optional<String> companyNumberInTransaction) {
+        return companyNumberInTransaction.isEmpty();
+    }
+
+    private boolean isRequestRoeUpdate(String companyNumberInScope, Optional<String> companyNumberInTransaction) {
+        boolean isValidOeNumberInScope = isValidOeNumber(companyNumberInScope);
+        var isValidOeNumberInTransaction = false;
```

Comment:
var isValidOeNumberInTransaction = companyNumberInTransaction.isPresent()? isValidOeNumber(companyNumberInTransaction.get()) : false;

Reply:
I see that this change has been made, but it's introduced a new code smell that needs addressing.